packages/core-ethereum/src/indexer/index.ts

```diff
@@ -448,6 +466,35 @@ class Indexer extends EventEmitter {
     }
   }
 
+  private async onTicketRedeemed(event: Event<'TicketRedeemed'>) {
+    if (Address.fromString(event.args.source).eq(this.address)) {
+      // the node used to lock outstandingTicketBalance
+      // rebuild part of the Ticket
+      const partialTicket: Partial<Ticket> = {
+        counterparty: Address.fromString(event.args.destination),
+        amount: new Balance(new BN(event.args.amount.toString()))
+      }
+      const outstandingBalance = await this.db.getPendingBalanceTo(partialTicket.counterparty)
+
+      try {
+        if (!outstandingBalance.toBN().gte(new BN('0'))) {
+          await this.db.resolvePending(partialTicket)
+        } else {
+          await this.db.resolvePending({
```

Comment:
this basically sets the pending balance to 0?

Reply:
either reduce by the ticket amount, or set to zero (when HoprDB somehow does not have all the pending tickets, e.g. being wiped in between)